src/domain/audience/create.js

```diff
@@ -0,0 +1,72 @@
+const ErrorHandler = require('../../infra/utils/errorHandler');
+const { Readable } = require('stream');
+const csv = require('csv-parser');
+const { Audience } = require('../../infra/database/models');
+const { utils, getDefaultProvider } = require('ethers');
+const config = require('./../../../config');
+const { v4: uuidv4 } = require('uuid');
+
+const provider = getDefaultProvider('homestead', {
+  etherscan: config.ETHERSCAN_API_KEY,
+});
+
+async function getMembersFromCSV(data) {
+  return new Promise((resolve, reject) => {
+    const stream = Readable.from(data);
+    let result = [];
+    stream
+      .pipe(csv())
+      .on('data', async (row) => {
+        const address = row.address;
+        if (utils.isAddress(address)) {
+          result.push({
+            ensName: '',
+            address: address,
+          });
+        } else if (address.includes('.eth')) {
+          result.push({
+            address: '',
+            ensName: address,
+          });
+        } else {
+          result.push({
+            ensName: '',
+            address: '',
+          });
+        }
+      })
+      .on('end', () => {
+        resolve(result);
+      })
+      .on('error', () => {
+        reject([]);
+      });
+  });
+}
+
+async function getAddressFromEnsName(members) {
+  for (let i = 0; i < members.length; ++i) {

```

Comment:
use promise and resolve instead of for loop. Better in terms of performance in JS.

So you can do something like this:

const allMemeberPromises = members.map(async () => { // whatever you have done in your for loop and return the full member object });

const members = await Promise.all(allMemeberPromises);